### DISCUSSION TITLE
Why no port to Chrome so far?

### DISCUSSION THREAD

**Discussion by conradRz** [tridactyl/tridactyl#4720] [Ideas]
Just wondering. 

Decided to ask here, as many of you know the code-base of this project well. I just have been using this extension for years.

Chrome has a way, way bigger "market" share, currently, and most extensions which work on Chrome also work on Edge (I know... that's probably not the target audience...) 

Something tells me that perhaps Google has made Chrome less mod-able, and this extension rather needs to change a lot browser-UI side?

**Comment by bovine3dom** [comment]
Tridactyl could work fine in Chrome (at least under manifest V2 which is end-of-life, but we'll need to port to manifest V3 at some point anyway for Firefox).

The blocker is that the small differences in the APIs between Firefox and Chrome - like the fact that Chrome doesn't have tab containers - mean that minor things that Tridactyl expects to exist don't, which then error out and stop the whole extension from working.

The solution I would like is for our tests to check that we are using APIs in a way that is safe for all the browsers we target. That should be easy because all the individual bits we need to do that already exist, it is just a case of wiring them up properly. But so far I have not managed to do it.

There's more info on this issue - https://github.com/tridactyl/tridactyl/issues/1619

**Reply by jman-schief** [reply]
> Tridactyl could work fine in Chrome (at least under manifest V2 which is end-of-life, but we'll need to port to manifest V3 at some point anyway for Firefox).

I'm curious about this statement: why porting to manifest V3 is a mandatory step for Tridactyl? To my knowledge, Firefox promises to support V3 but also support V2 ["for the forseeable future"](https://blog.mozilla.org/addons/2024/03/13/manifest-v3-manifest-v2-march-2024-update/).

Given the choice of Google to force their marketplace to adopt the controversial V3 (or GTFO), I see keeping V2 on Firefox as act of resistance :slightly_smiling_face:

**Reply by bovine3dom** [reply]
> Firefox, however, has no plans to deprecate MV2 and will continue to support MV2 extensions for the foreseeable future. And even if we re-evaluate this decision at some point down the road, we anticipate providing a notice of at least 12 months for developers to adjust accordingly and not feel rushed.

The only thing this concretely says is "we probably won't drop support for MV2 in the next 12 months", which sounds to me like they want to drop support for MV2 and will do eventually.

---

Planning not to deprecate MV2 is very different to not planning to

**Comment by n-ubo** [comment]
here is a small open source alternative if you're still interested (:
https://github.com/Nuboctane/clawjs
